docs/section5/Rule5-29_nonprimary.md

```diff
@@ -0,0 +1,38 @@
+# Envelope - Rule 5-29  
+**Schema Version** 0.0.23  
+**Primary Rule:** False
+**Rule ID:** 5-29  
+**Rule Description:** Baseline fenestration shall be assumed to be flush with the exterior wall, and no shading projections shall be modeled.
+**Appendix G Section:** Section G3.1-5(d) Building Modeling Requirements for the Baseline building  
+**Appendix G Section Reference:**  None  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:** None  
+
+**Evaluation Context:**  Each Data Element  
+**Data Lookup:** None  
+**Function Call:**
+  1. get_opaque_surface_type()
+  2. get_surface_conditioning_category()
+
+## Rule Logic:
+
+- Get surface conditioning category dictionary for B_RMR: ```scc_dictionary_b = get_surface_conditioning_category(B_RMR)```
+
+- For each building segment in the Baseline model: ```for building_segment_b in B_RMR.building.building_segments:```
+
+  - For each zone in building segment: ```for zone_b in building_segment_b.zones:```
+
+    - For each surface in zone: ```for surface_b in zone_b.surfaces:```
+
+      - Check if surface is above-grade wall or roof and is exterior: ```if ( get_opaque_surface_type(surface_b) == "ABOVE-GRADE WALL" ) AND ( scc_dictionary_b[surface_b.id] != "UNREGULATED" ):```
+
```

Comment:
description of line 28 says wall or roof but we check only for wall. I believe the description needs to be updated to remove 'roof'